syntax = "proto3";
package form_schema_version;




import "flowable_service_sdk/model/flowable_service/form_schema_version.proto";

/*
ListFormSchemaVersion请求
*/
message ListFormSchemaVersionRequest {
    /*
    表单实例Id
    */
    string formId = 1 ;
    /*
    页码
    */
    int32 page = 2 ;
    /*
    每页数据量
    */
    int32 pageSize = 3 ;
    /*
    表单名称
    */
    string name = 4 ;
    /*
    表单版本状态(unfinished 未完成, done已完成)
    */
    string state = 5 ;
}

/*
ListFormSchemaVersion返回
*/
message ListFormSchemaVersionResponse {
message List {
    /*
    表单版本信息
    */
    flowable_service.FormSchemaVersion versionInfo = 1 ;
    /*
    表单实例id
    */
    string instanceId = 2 ;
    /*
    表单名称
    */
    string name = 3 ;
    /*
    表单类别
    */
    string category = 4 ;
    /*
    表单作者
    */
    string creator = 5 ;
    /*
    表单说明
    */
    string memo = 6 ;
    /*
    创建时间
    */
    string ctime = 7 ;
}
        
    
    /*
    页数
    */
    int32 page = 1 ;
    /*
    页大小
    */
    int32 page_size = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    数据列表
    */
    repeated List list = 4 ;
}

/*
ListFormSchemaVersionApi返回
*/
message ListFormSchemaVersionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    form_schema_version.ListFormSchemaVersionResponse data = 4 ;
}

